templates/nextflow-s3-template.yaml

```diff
@@ -95,62 +98,89 @@ Resources:
               - !Sub arn:aws:s3:::${Bucket}
               - !Sub arn:aws:s3:::${Bucket}/*
 
-          - Sid: ReadWriteAccess
-            # gives grantees read/write write access to the bucket with required --acl
-            Effect: Allow
-            Principal:
-              AWS: !Split
-                - ","
-                - !Join
-                    - ","
-                    - - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
-                      - !Join
-                          - ","
-                          - !Ref ReadWriteAccessArns
-            Action:
-              - "s3:*Object*"  
-            Condition:
-              StringEquals:
-                s3:x-amz-acl: bucket-owner-full-control
-            Resource:
-              - !Sub arn:aws:s3:::${Bucket}/*    
+          - !If
+            - NoReadWriteAccessArns
+            - Sid: ForgeUserReadWriteAccess
+              # gives forge user read/write write access to the bucket if no read/write arns are supplied
+              Effect: Allow
+              Principal:
+                AWS:
+                  - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
+              Action:
+                - "s3:*Object*"  
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
```

Comment:
We don't need this statement condition for the Tower Forge user because it's an IAM principal within the account and therefore the object ownership issue wouldn't apply if it uploaded objects to the bucket. In fact, this might break things because Nextflow probably doesn't use the `bucket-owner-full-control` ACL. So, it's actually a good thing that we have different statements for the Tower Forge user and other human users/roles.